doc/developers/contributing.rst

```diff
@@ -389,6 +390,28 @@ and Cython optimizations.
      [doc build]            Docs built including example gallery plots
      ====================== ===================
 
+Stalled pull requests
+......................
+
+As contributing a new feature can be a lengthy process, some
+pull requests appear inactive but unfinished. In such a case, taking
+them over is a great service for the project.
+
+A good etiquette for this is:
+
+* To decide whether a PR is stalled, asking the contributor if she/he
+  plans to continue working on the PR in the near future. Failure to
+  respond within a week implies that the PR is stalled.
+
+  Note that if the PR has received earlier comments on the
```

Comment:
How's this:

---
A good etiquette for this is:  

To determine whether a PR is active or stalled, ask the contributor if they plan to work on the PR in the future.  

An example is:
>It would be great if you could follow up on the work that you started.  Please indicate if you will continue to work on it.  Refer to the **Stalled pull requests** section
in the [Contributing Document](https://scikit-learn.org/stable/developers/contributing.html).

A "dynamic activity" is defined as one which moves the PR forward such as a commit or reply to a question.  A "static activity" is one which does not, such as a status update.  

Failure to respond within a week with a dynamic activity implies that the PR is stalled.

​For PRs which have not shown any type of activity for at least a month, the ​allotted timeframe for a dynamic activity is three days.

For PRs which are submitted during a sprint, it is expected that dynamic activity will take place during the sprint duration.  Follow-up time frames post-sprint may be determined and adjusted by the sprint organizers.

Reply:
I was more concerned about the possibility of picking up a stalled PR **during** a sprint, not about stalled PR that were submitted by sprint attendee (even though that's also an issue).

Sticking to the current text, sprint attendees would have to wait one day at least for the original contributor's response before working on a stalled PR, which doesn't make sense since sprints are often one-day long.

Reply:
how would we know if a PR is stalled *during* a sprint?

Reply:
I am not talking about PRs stalled during a sprint, I am talking about stalled PRs that sprint attendees pick up and finish.

During the WIMLDS sprint you organized where I helped as a TA, @amueller  had prepared a list of stalled PRs that attendees could pick up. I am talking about those PRs. All I'm saying is, if it's for a sprint, we should be allowed to bypass the x-days wait.

Reply:
Another alternative, which makes it unnecessary to have an "aggressive" approach, is for us to actually go through the old PRs continuously and add "stalled" or "help wanted" tags as appropriate. Some of that can be automated with a bot working on stalled PRs [semi]automatically. I suppose if we actually have those tags attached to enough PRs, there would be no need to speed up the process and there will always be work to pick up.

Reply:
I think it's worthwhile to have a standard for giving notice of activity.

If the default is set in such a way, (if there is no activity, it will be considered stalled), then it encourages lack of communication.

I send multiple reminders to our meetup attendees who have RSVP'd for our events, and 2/3 of them ignore it, so not sure if collective behavior can be changed.  
And the majority of the WiMLDS sprint participants did not complete their PRs or respond to multiple follow-up communications.  

If the standard for communication had been set sooner (or had I known how things would have worked out), we would have been about 2 months further ahead in merging these PRs.

Reply:
Oh I didn't mean thta we should not have that notice period. I just meant that if we actively try to keep an eye on the stalled ones, then the notice period can be longer (I really think except for urgent cases, 3 days (or even a week) is too short). In most cases, unless we're close to a release, it really doesn't matter if it takes a week for a PR to get in or a month. The crucial point is to keep them going, I guess.

Reply:
How about giving 2 weeks, and leave sprint PRs up to the discretion of the organizers?  They can communicate follow-up information during the sprint in person.

Reply:
I like that. It'd be nice if organizers tag those PRs with a "Sprint" tag.

Reply:
How about (that keeps it quite simple): 

>Failure to respond within 2 weeks with an activity that moves the PR forward will result in tagging that PR with "help wanted".

>Follow-up for un-merged sprint PRs will be communicated to participants at the sprint, and those PRs will be tagged "sprint"

Reply:
@GaelVaroquaux This works.  I agree, shorter is better.

Reply:
Thanks a lot for this fruitful discussion.